### PR TITLE
Promote wins_dns_zone Module from community

### PR DESCRIPTION
**Pull request by Yaish25491**
Updating tasks to use modules that migrated to Windows.ad repo.

##### SUMMARY

Updating tasks to use modules that migrated to Windows.ad repo.
Verifying modules on Windows Server 2019 and Windows Server 2022

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
win_dns_zone.ps1
win_dns_zone.py
win_dns_zone/aliases
win_dns_zone/defaults/main.yml
win_dns_zone/tasks/activedirectory.yml
win_dns_zone/tasks/main.yml
win_dns_zone/tasks/standalone.yml

##### ADDITIONAL INFORMATION
Migrating Win_dns_zone from community to ansible.windows

```

### CODE REVIEW

File: plugins/modules/win_dns_zone.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      - Required if l(type=secondary), l(type=forwarder) or l(type=stub),
+        otherwise ignored.
+      - At least one server is required.
+    elements: str
+    type: list
+'''
+
+EXAMPLES = r'''
+- name: Ensure primary zone is present
+  community.windows.win_dns_zone:
```
Comment (jborean93) [inline plugins/modules/win_dns_zone.py:88]: All the examples need to change from `community.windows.win_dns_zone` to `ansible.windows.win_dns_zone`.

File: tests/integration/targets/win_dns_zone/aliases [hunk 2]
```diff
@@ -0,0 +1 @@
+shippable/windows/group1
```
Comment (jborean93) [inline tests/integration/targets/win_dns_zone/aliases:1]: Please change to group 3. Based on the nightly run it is the group with the smallest runtime https://dev.azure.com/ansible/ansible.windows/_build/results?buildId=128352&view=results.

File: tests/integration/targets/win_dns_zone/tasks/activedirectory.yml [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+  ansible.windows.win_feature:
+    name:
+      - AD-Domain-Services
+      - DNS
+    include_management_tools: true
+    include_sub_features: true
+    state: present
+
+- name: Ensure domain is present
+  ansible.windows.win_domain:
```
Comment (jborean93) [inline tests/integration/targets/win_dns_zone/tasks/activedirectory.yml:12]: We should try and use `microsoft.ad.domain` module here. I'll try and submit a PR to add it as a test requirement as it is somewhat complicated to get it set up.

Comment (jborean93) [reply tests/integration/targets/win_dns_zone/tasks/activedirectory.yml:12]: https://github.com/ansible-collections/ansible.windows/pull/691 has been merged so now `microsoft.ad` is available in the integration tests.

File: tests/integration/targets/win_dns_zone/aliases (hunk 2 above)
Comment (Yaish25491) [reply tests/integration/targets/win_dns_zone/aliases:1]: Should we apply this change to all modules?

File: tests/integration/targets/win_dns_zone/tasks/activedirectory.yml (hunk 3 above)
Comment (Yaish25491) [reply tests/integration/targets/win_dns_zone/tasks/activedirectory.yml:12]: Done.

**Pull request merged**